runtime/test/manifest-test.js

```diff
@@ -1409,4 +1409,33 @@ resource SomeName
     assert(slotConnection._providedSlots.provideSlot);
     assert.equal(slotConnection._providedSlots.provideSlot.sourceConnection, slotConnection);
   });
+
+  it('can parse particle arguments with tags and optional names', async () => {
+    let manifest = await Manifest.parse(`
+      schema Dog
+      schema Sled
+      schema DogSled
+      particle DogSledMaker in 'thing.js'
+        in Dog #leader
```

Comment:
Could be worth adding a case with no explicit name and two tags and validating only the first tag is chosen.

Reply:
Ack, I'll add some more tests in my follow on work.